src/server/routes/api.js

```diff
@@ -222,6 +223,40 @@ module.exports = function(app) {
 
     });
 
+  api.post('/admin/events', requireAuth, 
+    roleAuth(roles.ROLE_ADMIN), upload.single('logo'), (req, res) => {
+      let event = new Event;
+      const {closeTimeDay, closeTimeMonth, closeTimeYear} = req.body;
+
+      ['closeTimeDay', 'closeTimeMonth', 'closeTimeYear', 'logo'].forEach(k => delete req.body[k]);
+
+      req.body.closeTime = closeTimeYear + '-' +
+      closeTimeMonth.padStart(2, '0') + '-' +
+      closeTimeDay.padStart(2, '0')
+      + 'T00:00:00.000Z';
+
+      Object.entries(req.body).forEach(([k, v]) => event[k] = v);
+
+      if (['Admin', 'Member'].includes(req.user.role)) {
+        event.organisers = [req.user._id]
```

Comment:
We're already requiring roleAuth(roles.ROLE_ADMIN), so we can safely assume that whoever created the account should be the organiser.

Reply:
This bit actually adds them to the event as an organiser - it's not an authentication thing

Reply:
I meant for Line 240 - The conditional checks whether the user has the correct role. Are you preventing sponsors from creating events? In that case, only Admins and above should be listed here. You can do 
`getRole(req.user.role) >= getRole(Roles.ROLE_ADMIN)`

Reply:
Developers shouldn't be added to the organisers array because they see everything by default anyway - that's what I'm checking for here

Reply:
Please use `getRole(req.user.role) == getRole(Roles.ROLE_ADMIN)` as it uses the enumeration value of the role rather than its name.